controller/profiling.go

```diff
@@ -0,0 +1,24 @@
+package controller
```

Comment:
I will caveat that I'm not a pprof expert. This server config worked in my testing on a local image but I'm certainly open to feedback.

I also didn't include any tests as the functionality here is much simpler than the metrics server for example. However, maybe I could add something similar to the healthz server setup to test if the server returns responses?

Reply:
I am ok without testing for this.